crates/air-lib/trace-handler/src/data_keeper/trace_slider.rs

```diff
@@ -62,9 +64,10 @@ impl TraceSlider {
         Some(result)
     }
 
-    pub(crate) fn set_position_and_len(&mut self, position: usize, subtrace_len: usize) -> KeeperResult<()> {
+    pub(crate) fn set_position_and_len(&mut self, position: TracePos, subtrace_len: usize) -> KeeperResult<()> {
+        let pos: usize = position.into();
         // it's possible to set empty subtrace_len and inconsistent position
-        if subtrace_len != 0 && position + subtrace_len > self.trace.len() {
+        if subtrace_len != 0 && pos + subtrace_len > self.trace.len() {
```

Comment:
Let's implement `cheched_add` for `TracePos` instead.

Reply:
Would you like to .unwrap() it or bubble as error?

Reply:
Will be resolved in the #275.